client/src/app/core/repositories/management/meeting-repository.service.ts

```diff
@@ -249,14 +227,35 @@ export class MeetingRepositoryService extends BaseRepository<ViewMeeting, Meetin
         return temp;
     }
 
-    private getUserGroupUpdatePayload(
-        ids: Id[] = [],
+    /**
+     * Maps group-ids for users to their id.
+     * If the given groupId is already existing, it will be removed. Otherwise it will be added.
+     *
+     * @param previousData An object containing the previous group-ids for users mapped to user-ids
+     * @param users An array of users, whose groups for a specific meeting are updated
+     * @param meetingId The id of a meeting, users assigned to or removed from
+     * @param groupId The id of a group, that is assigned to or removed from a user
+     *
+     * @returns An object containing user-ids as keys and an array of ids as values, which are the next group-ids
+     * for the related user.
+     */
+    private getNewGroupsForUsers(
+        previousData: { [userId: number]: Id[] },
+        users: ViewUser[],
         meetingId: Id,
-        groupId?: Id
-    ): { id: Id; group_$_ids: { [meetingId: number]: Id[] } }[] {
-        return ids.map(id => ({
-            id,
-            group_$_ids: { [meetingId]: groupId ? [groupId] : [] }
-        }));
+        groupId: Id
+    ): { [userId: number]: Id[] } {
+        const nextData = { ...previousData };
```

Comment:
why do you always copy? You can do this as an side effect (so no return)

Reply:
Yes, it is because I don't like side effects so much... But I can change it.